test/unit/kernel/CMakeLists.txt

```diff
@@ -58,6 +58,13 @@ SET(_SOURCE_DIR "src/")
 # Add all the source files in all recursive subdirectories and group them accordingly.
 append_recursive_files_add_to_src_group("${_SOURCE_DIR}" "${_SOURCE_DIR}" "cpp" _FILES_SOURCE)
 
+# NVCC does not incorporate the COMPILE_OPTIONS of a target but only the CMAKE_CXX_FLAGS
+IF(ALPAKA_ACC_GPU_CUDA_ENABLE AND NOT ALPAKA_CUDA_COMPILER MATCHES "clang")
```

Comment:
why do we add this not to `ALPAKA_ADD_EXECUTABLE`?

Reply:
and maybe also to `ALPAKA_ADD_LIBRARY`?

Reply:
The problem is that we would need to know the target dependencies within those functions to correctly add at least the `COMPILE_OPTIONS` at the moment of the `ALPAKA_ADD_EXECUTABLE` call. However, the dependencies are in general only added after creating the target which is too late for FindCUDA.
We could add a convenience function that wraps those five lines. However, I intend to remove all those occurences with a single one within a follow-up PR. Furthermore, this only works because we know that `common` does not have any transitive dependency which defines `COMPILE_OPTIONS`. To handle this correctly, we would need to know transitive dependencies of the given targets and get the properties for all of them. I do not know how to do this correctly.

Reply:
Maybe we could work with generator expressions for `ALPAKA_ADD_EXECUTABLE`/`_LIBRARY`?

Reply:
FindCUDA does not support any such thing. It really only supports those `COMPILE_OPTIONS` which are part of `CMAKE_CXX_FLAGS` at the time `cuda_add_executable` is called. So nothing we could do.

Reply:
We really need to add Clang -x cuda to CMake so we can switch to the CUDACXX language feature and get rid of `FindCUDA` wrappers.